agent/proxycfg/connect_proxy.go

```diff
@@ -614,3 +621,66 @@ func (s *handlerConnectProxy) handleUpdate(ctx context.Context, u UpdateEvent, s
 	}
 	return nil
 }
+
+// hcpMetricsConfig represents the basic opaque config values for pushing telemetry to HCP.
+type hcpMetricsConfig struct {
+	// HCPMetricsBindSocketDir is a string that configures the directory for a
+	// unix socket where Envoy will forward metrics. These metrics get pushed to
+	// the HCP Metrics collector to show service mesh metrics on HCP.
+	HCPMetricsBindSocketDir string `mapstructure:"envoy_hcp_metrics_bind_socket_dir"`
+}
+
+func parseHCPMetricsConfig(m map[string]interface{}) (hcpMetricsConfig, error) {
+	var cfg hcpMetricsConfig
+	err := mapstructure.WeakDecode(m, &cfg)
+
+	if err != nil {
+		return cfg, fmt.Errorf("failed to decode: %w", err)
+	}
+
+	return cfg, nil
+}
+
+// maybeInitializeHCPMetricsWatches will initialize a synthetic upstream and discovery chain
+// watch for the HCP metrics collector, if metrics collection is enabled on the proxy registration.
+func (s *handlerConnectProxy) maybeInitializeHCPMetricsWatches(ctx context.Context, snap ConfigSnapshot) error {
+	hcpCfg, err := parseHCPMetricsConfig(s.proxyCfg.Config)
+	if err != nil {
+		s.logger.Error("failed to parse connect.proxy.config", "error", err)
+	}
+
+	if hcpCfg.HCPMetricsBindSocketDir != "" {
+		if !strings.HasSuffix(hcpCfg.HCPMetricsBindSocketDir, "/") {
+			hcpCfg.HCPMetricsBindSocketDir += "/"
+		}
+
+		// The path includes the proxy ID so that when multiple proxies are on the same host
+		// they each have a distinct path to send their metrics.
+		path := fmt.Sprintf("%s%s_%s.sock", hcpCfg.HCPMetricsBindSocketDir, s.proxyID.NamespaceOrDefault(), s.proxyID.ID)
+
+		upstream := structs.Upstream{
+			DestinationNamespace: acl.DefaultNamespaceName,
+			DestinationPartition: s.proxyID.PartitionOrDefault(),
+			DestinationName:      api.HCPMetricsCollectorName,
```

Comment:
The collector will be deployed in the default namespace of each admin partition using the well-known `api.HCPMetricsCollectorName`